### ISSUE TITLE
[question]  permission revoke for app invocation

### ISSUE THREAD

**Issue by klearnstodev** [OwnGoalStudio/NoRedirect#3]
I'm really glad to see the first tweak related to app invocation in the community!

I have a question that's been bothering me for a long time, so I hope I can get some help here.

In iOS, when app X calls app Y for the first time, there's a dialog that asks the user whether they allow this call. The issue is, this permission is "once for all," meaning that once you allow it, it won't ask again in the future.

Here's the problem: if I accidentally hit 'allow' during that first prompt, where can I find the option to revoke that permission? Specifically, how can I reset the calling relationship between app X and app Y so that the next time app X calls app Y, the permission prompt will appear again? This information should be stored in a database somewhere outside of the sandbox, right?

I've tested deleting and reinstalling both apps, but that doesn’t seem to work. Resetting the system settings would definitely do it, though.

Of course, your tweak should be able to solve this problem, but using it just to prevent specific calls feels a bit overkill.

Thanks in advance!

**Comment by Lessica** [comment]
The question is, if you hit “Cancel” or “Disallow” on that dialog, the app usually opens Safari, App Store or a web page as an alternative.

“No Redirect” can give a _false positive_ to the source app and is better for ads cancellation.

And there is a “Redirect History”. You may check what's going on later.

**Comment by klearnstodev** [comment]
> The question is, if you hit “Cancel” or “Disallow” on that dialog, the app usually opens Safari, App Store or a web page as an alternative.

I totally agree with that, especially when it comes to some of Chinese apps from the same group or affiliate.

> “No Redirect” can give a _false positive_ to the source app and is better for ads cancellation.
> 
> And there is a “Redirect History”. You may check what's going on later.

Although my question was not discussed in detail, that's okay. Let's keep it open to see if any other users in the community have any insights on this point. Thanks for the comments from the developers. It is really a nice tweak.
